### PR TITLE
Python ADO packaging pipeline

### PR DESCRIPTION
**Pull request by jchen351**
This pull request includes changes to several files in the project. The changes are mainly related to the build and packaging configurations for the project. The most important changes include:

Changes to the build process:
* [`.github/workflows/linux-cpu-arm64-build.yml`](diffhunk://#diff-41c373598f771a9d90102d4d380ba704651d296b865d6974fe3bbd7d019b35c1R39-R46): Updated the build process to use a new Docker image and added Azure login commands. [[1]](diffhunk://#diff-41c373598f771a9d90102d4d380ba704651d296b865d6974fe3bbd7d019b35c1R39-R46) [[2]](diffhunk://#diff-41c373598f771a9d90102d4d380ba704651d296b865d6974fe3bbd7d019b35c1L53-R55)
* [`.github/workflows/linux-cpu-x64-build.yml`](diffhunk://#diff-e758ed36b14737d7c89353329cb735a6d081f5fa288b809d49e47bfc0ade96c3L17-L18): Removed unnecessary lines in the build process.
* [`.github/workflows/linux-gpu-x64-build.yml`](diffhunk://#diff-e941545683ca21b146fc6f414d9fe9e311f5f5efc44fce47287e8b0d835434aeR39-R59): Updated the build process to use a new Docker image and added Azure login commands. [[1]](diffhunk://#diff-e941545683ca21b146fc6f414d9fe9e311f5f5efc44fce47287e8b0d835434aeR39-R59) [[2]](diffhunk://#diff-e941545683ca21b146fc6f414d9fe9e311f5f5efc44fce47287e8b0d835434aeL67-R69)

Changes to the packaging configurations:
* [`.pipelines/pypl-publishing.yml`](diffhunk://#diff-e7c846e17ab5319529d839ddce86308159f05bc27ed172b501c1b996d00d838cL2-R49): Updated the packaging configurations for different platforms and restructured the stages in the pipeline.
* [`.pipelines/stages/jobs/py-linux-cpu-packaging-job.yml`](diffhunk://#diff-da741590402f2646d68481d93e55f810d2140147c06942438ca8ce480273dbceR1-R136): Added new parameters and updated the packaging job for Linux CPU.
* [`.pipelines/stages/jobs/py-linux-gpu-packaging-job.yml`](diffhunk://#diff-f5e402bc55a3463f44b0b0a5d3efba72dd9e9a549fb2bdf9fcf8857008f6a2acR1-R135): Added new parameters and updated the packaging job for Linux GPU.

Other changes:
* [`.gdn/.gdntsa`](diffhunk://#diff-b4fbb94e5e0cd5b2aae8fa62763e8099598170a5478d9e610bdced711de09de1R1-R3): Added a new codebase name.

### CODE REVIEW

File: .pipelines/stages/jobs/py-linux-cpu-packaging-job.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    displayName: Unzip OnnxRuntime
+
+  - bash: |
+      mv  onnxruntime-linux-$(buildArch)-$(ort_version) ort
+    displayName: Rename Onnxruntime to ort
+    workingDirectory: '$(Build.SourcesDirectory)/onnxruntime-genai'
+
+  - script: |
+      set -e -x
+      az login --identity -u /subscriptions/00c06639-6ee4-454e-8058-8d8b1703bd87/resourcegroups/proj-onnxruntime-dev/providers/Microsoft.ManagedIdentity/userAssignedIdentities/ortcibuild_readonly
```
Comment (snnn) [inline .pipelines/stages/jobs/py-linux-cpu-packaging-job.yml:64]:
```
az login --identity -u 63b63039-6328-442f-954b-5a64d124e5b4
```

Use an ID is better. Because an ID doesn't get changed if the resource is moved to another resource group.

File: tools/ci_build/github/linux/docker/manylinux/scripts/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+numpy==1.21.6 ; python_version < '3.11'
+numpy==1.24.2 ; python_version == '3.11'
+numpy==1.26.0 ; python_version >= '3.12'
+mypy
+pytest
+setuptools>=68.2.2
+wheel
+onnx==1.15.0
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/manylinux/scripts/requirements.txt:8]: Do we need to install onnx? If yes, try to pin it to the latest version.

File: .pipelines/stages/jobs/py-linux-cpu-packaging-job.yml (hunk 1 above)
Comment (jchen351) [reply .pipelines/stages/jobs/py-linux-cpu-packaging-job.yml:64]: Tried but it didnt work
https://aiinfra.visualstudio.com/ONNX%20Runtime/_build/results?buildId=418186&view=logs&j=e8617949-7cc5-55ad-1f2e-d6930d975d4f&t=1fd12ec2-da31-59e4-39ed-d2b202af722b&l=12

Comment (snnn) [reply .pipelines/stages/jobs/py-linux-cpu-packaging-job.yml:64]: Because you missed one digit when you copy-paste!

File: .gdn/.gdntsa [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (snnn) [inline .gdn/.gdntsa:1]: Replace references of this file with ".config/tsaoptions.json". Then delete this file. You may do it in another PR.

**Review by snnn — APPROVED** [review]
